rfai/service/rfai_service.py

```diff
@@ -20,22 +20,55 @@ def __init__(self, repo):
         self.stake_dao = StakeDAO(repo=repo)
         self.foundation_member_dao = FoundationMemberDAO(repo=repo)
 
-    def validate_request_filter_parameters(self, params):
-        for param in params:
-            if param in ['requester', 'status', 'request_id']:
-                pass
-            else:
-                raise Exception("Bad query parameters.")
+    def format_filter_params(self, query_parameters):
+        filter_params = {}
+        if "status" in query_parameters.keys():
+            filter_params["status"] = query_parameters["status"]
+        elif "requester" in query_parameters.keys():
+            filter_params["requester"] = query_parameters["requester"]
+        elif "request_id" in query_parameters.keys():
+            filter_params["request_id"] = query_parameters["request_id"]
+        return filter_params
 
     def get_requests(self, query_string_parameters):
         status = query_string_parameters.get("status", None)
-        self.validate_request_filter_parameters(params=query_string_parameters.keys())
+        filter_parameter = self.format_filter_params(query_parameters=query_string_parameters)
         if status is not None and status.upper() in RFAIStatusCodes.__members__:
-            query_string_parameters["status"] = RFAIStatusCodes[status.upper()].value
-        else:
-            raise Exception("Invalid request status.")
-        requests_data = self.request_dao.get_request_data_for_given_requester_and_status(
-            filter_parameter=query_string_parameters)
+            status_code = RFAIStatusCodes[status.upper()].value
+            query_string_parameters["status_code"] = status_code
+            current_block_no = obj_blockchain_utils.get_current_block_no()
+
+            if status_code == RFAIStatusCodes.APPROVED_ACTIVE.value:
+                requests_data = self.request_dao.get_approved_active_request(current_block_no=current_block_no,
+                                                                             filter_parameter=filter_parameter)
+
+            elif status_code == RFAIStatusCodes.APPROVED_SOLUTION_VOTE.value:
+                requests_data = self.request_dao.get_approved_solution_vote_request(current_block_no=current_block_no,
+                                                                                    filter_parameter=filter_parameter)
```

Comment:
read about factory pattern and strategy pattern how we can eliminate this if else